prow/gerrit/reporter/reporter.go

```diff
@@ -159,36 +158,35 @@ func (c *Client) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
 	selector := labels.Set{
 		clientGerritRevision: pj.ObjectMeta.Labels[clientGerritRevision],
 		pjTypeLabel:          pj.ObjectMeta.Labels[pjTypeLabel],
+		gerritReportLabel:    pj.ObjectMeta.Labels[gerritReportLabel],
 	}
+
+	var toReportJobs []*v1.ProwJob
 	if pj.ObjectMeta.Labels[gerritReportLabel] == "" {
-		// Shouldn't happen, adapter should already have defaulted to Code-Review
-		logger.Errorf("Gerrit report label not set for job %s", pj.Spec.Job)
-	} else {
-		selector[gerritReportLabel] = pj.ObjectMeta.Labels[gerritReportLabel]
-	}
+		toReportJobs = append(toReportJobs, pj)
+	} else { // generate an aggregated report
 
-	// list all prowjobs in the patchset matching pj's type (pre- or post-submit)
+		// list all prowjobs in the patchset matching pj's type (pre- or post-submit)
 
-	pjsOnRevisionWithSameLabel, err := c.lister.List(selector.AsSelector())
-	if err != nil {
-		logger.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
-		return nil, err
-	}
+		pjsOnRevisionWithSameLabel, err := c.lister.List(selector.AsSelector())
+		if err != nil {
+			logger.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
+			return nil, err
+		}
 
-	// generate an aggregated report:
-	var toReportJobs []*v1.ProwJob
-	mostRecentJob := map[string]*v1.ProwJob{}
-	for _, pjOnRevisionWithSameLabel := range pjsOnRevisionWithSameLabel {
-		job, ok := mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job]
-		if !ok || job.CreationTimestamp.Time.Before(pjOnRevisionWithSameLabel.CreationTimestamp.Time) {
-			mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = pjOnRevisionWithSameLabel
+		mostRecentJob := map[string]*v1.ProwJob{}
+		for _, pjOnRevisionWithSameLabel := range pjsOnRevisionWithSameLabel {
+			job, ok := mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job]
+			if !ok || job.CreationTimestamp.Time.Before(pjOnRevisionWithSameLabel.CreationTimestamp.Time) {
+				mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = pjOnRevisionWithSameLabel
+			}
 		}
-	}
-	for _, pjOnRevisionWithSameLabel := range mostRecentJob {
-		if pjOnRevisionWithSameLabel.Status.State == v1.AbortedState {
-			continue
+		for _, pjOnRevisionWithSameLabel := range mostRecentJob {
+			if pjOnRevisionWithSameLabel.Status.State == v1.AbortedState {
+				continue
```

Comment:
Note that if the report label is empty, we no longer avoid reporting aborted jobs. Is that behavior intentional?

Reply:
Ahh, fixed. Thanks!